src/Cardano/Wallet.hs

```diff
@@ -301,22 +305,23 @@ walletNameMinLength = 1
 walletNameMaxLength :: Int
 walletNameMaxLength = 255
 
-newtype WalletId = WalletId Text
-    deriving (Eq, Ord, Show)
+newtype WalletId = WalletId UUID
+    deriving (Generic, Eq, Ord, Show)
 
-newtype WalletTimestamp = WalletTimestamp Microsecond
-    deriving (Eq, Ord, Show)
-
-data WalletStatus
+data WalletState
     = Ready
-    | Restoring
-    deriving (Eq, Show)
+    | Restoring !(MeasuredIn "percent" Word)
```

Comment:
Hm… You have just defined Aeson instances on a `Cardano.Wallet` type, haven't you? :)

Reply:
Ok, not really, but I'm confused by this 🤔 

Why have `ApiT` at all?

Reply:
> Why have `ApiT` at all?

Having `ApiT` allows us to avoid orphan instances, and to keep the following concerns separate:
1. How we define a type that we use internally.
2. How we represent values of that type on the wire.

For types that are only ever seen in the API, we can define `FromJSON` and `ToJSON` instances directly for those types, and the job is done.

For types that appear in the API, but are actually defined *elsewhere* (like `AddressPoolGap`), we still want to define `FromJSON` and `ToJSON` instances in the same way as the other types, but we don't want to place those instances in `Primitive` (to keep the separation of concerns).

Therefore, we need to wrap such types in a `newtype` wrapper, in order to be able to give them non-orphan instances. We could, of course, use a different `newtype` for each and every such type. But in the end, most of these wrappers would be the same. Hence we define `ApiT` as a common wrapper, and give each `ApiT a` a unique instance of `FromJSON` and `ToJSON`.